lib/msal-react-native-android-poc/MSALModule/android/src/main/java/com/reactlibrary/MSALModule.java

```diff
@@ -175,6 +175,49 @@ public void onError(@NonNull MsalException exception) {
         });
     }
 
+    /**
+     * acquireToken: attempts to obtain a token interactively
+     * Parameters: string scopesValue (string containing scopes separated by a " "), Promise promise (returned to an async function)
+     */
+    @ReactMethod
+    public void acquireToken(String scopesValue, Promise promise) {
+        if (!scopesValue.isEmpty()) {
+            publicClientApplication.acquireToken(getCurrentActivity(), scopesValue.toLowerCase().split(" "), getAuthInteractiveCallback(promise));
+        } else {
+            //the default is User.Read
+            String[] array = {"User.Read"};
+            publicClientApplication.acquireToken(getCurrentActivity(), array, getAuthInteractiveCallback(promise));
+        }
+    }
+
+    /**
+     * getAuthInteractiveCallback is the callback function for acquireToken.
+     * If successful, it will return a map of the IAuthenticationResult.
+     * If there's an error, the exception will be returned.
+     */
+
+    private AuthenticationCallback getAuthInteractiveCallback(final Promise promise) {
+        return new AuthenticationCallback() {
+            @Override
+            public void onCancel() {
```

Comment:
Is this event raised by MSAL android or the end-user? We have cleanup code for things like cache when a login event is cancelled, you'll want to make sure this is still being done.

Reply:
I'm waiting from a response from the android team regarding cleanup code, but I believe the event is raised by the end user (when they back out of the auth flow).  Looking at their sample app, all they have within their onCancel is a logging statement. So hopefully I won't have to worry about clearing the cache but I'll check to make sure.

Reply:
Shane says that MSAL Android doesn't create anything in the token cache until there is an actual token response. So when the user cancels it wouldn't have gotten that far yet